sentry-log4j2/src/test/kotlin/io/sentry/log4j2/SentryAppenderTest.kt

```diff
@@ -0,0 +1,287 @@
+package io.sentry.log4j2
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryLevel
+import io.sentry.core.transport.ITransport
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.apache.logging.log4j.ThreadContext
+import org.apache.logging.log4j.core.LoggerContext
+import org.apache.logging.log4j.core.config.AppenderRef
+import org.apache.logging.log4j.core.config.Configuration
+import org.apache.logging.log4j.core.config.LoggerConfig
+import org.awaitility.kotlin.await
+
+class SentryAppenderTest {
+    private class Fixture(minimumBreadcrumbLevel: Level? = null, minimumEventLevel: Level? = null) {
+        val transport = mock<ITransport>()
+        val logger: Logger
+        val loggerContext = LogManager.getContext() as LoggerContext
+
+        init {
+            loggerContext.start()
+            val config: Configuration = loggerContext.configuration
+            val appender = SentryAppender("sentry", null, "http://key@localhost/proj", minimumBreadcrumbLevel, minimumEventLevel, transport)
+            config.addAppender(appender)
+
+            val ref = AppenderRef.createAppenderRef("sentry", null, null)
+
+            val loggerConfig = LoggerConfig.createLogger(false, Level.TRACE, "sentry_logger", "true", arrayOf(ref), null, config, null)
+            loggerConfig.addAppender(appender, null, null)
+            config.addLogger(SentryAppenderTest::class.java.name, loggerConfig)
+
+            loggerContext.updateLoggers(config)
+
+            appender.start()
+            loggerContext.start()
+
+            logger = LogManager.getContext().getLogger(SentryAppenderTest::class.java.name)
+        }
+    }
+
+    private lateinit var fixture: Fixture
+
+    @AfterTest
+    fun `stop log4j2`() {
+        fixture.loggerContext.stop()
+    }
+
+    @BeforeTest
+    fun `clear MDC`() {
+        ThreadContext.clearAll()
+    }
+
+    @Test
+    fun `converts message`() {
+        fixture = Fixture(minimumEventLevel = Level.DEBUG)
```

Comment:
I'd use this type of params in the `getSut` method as all the other tests, looks like we have a different type of Fixture way of doing now.

Reply:
I am not sure how this can be done as in case of this test as this is an integration test and the appender that are actually testing is not used directly in tests. Could you make a PR to this PR so that I can get a better feeling of what you have in mind?

Reply:
I see, that's the problem, we are testing the appender but we don't have any references to it, its more like an integration test, where the appender gets created as you were setting the logger for real, what I meant is something like:

https://github.com/getsentry/sentry-android/blob/master/sentry-android-timber/src/test/java/io/sentry/android/timber/SentryTimberIntegrationTest.kt#L18-L32

but I feel that creating a Fixture every time looks different than all the other tests, usually, we create a `sut` per test but not a fixture.
maybe our `sut` should be the `logger` then as it is the one that makes the real calls.

@bruno-garcia introduced the fixture pattern in our tests, maybe he has something to share

Reply:
Usually with this pattern Fixture doesn't vary (so can be instantiated with the test class) but its properties are mutable to set the state you need by the time you call `getSut()` in the test body.

Would also prefer to stick to that approach if possible but TBQH no strong feelings here, whatever you folks agree is fine by me

Reply:
I am happy to change it but really I do not see how. PRs welcome :-)

Reply:
Ok Bruno explained it to me :-)